src/util/getStep.js

```diff
@@ -1,18 +1,54 @@
+const getPixels = require('get-pixels');
 module.exports = {
     getPreviousStep : function () {
         return this.getStep(-1);
     },
-    
+
     getNextStep : function() {
         return this.getStep(1);
     },
-    
+
     getInput : function(offset){
         if(offset + this.getIndex() === 0) offset++;
         return this.getStep(offset - 1).output;
     },
-    
+
     getOuput : function(offset){
         return this.getStep(offset).output;
+    },
+
+    getOptions : function(){
+        return this.getStep(0).options;
+    },
+
+    setOptions : function(optionsObj){
+        let options = this.getStep(0).options;
+        for(let key in optionsObj){
+            if(options[key]) options[key] = optionsObj[key];
+        }
+    },
+
+    getFormat : function(){
+        return this.getStep(-1).output.format;
+    },
+
+    getHeight : function(callback){
+        getPixels(this.getStep(-1).output.src,function(err,pixels){
```

Comment:
would using `getPixels` require parsing the whole image? I wonder if that'd be an expensive thing to do and if there are any reasonable alternatives. What do you think?

Reply:
Hmm...yeah now that you have mentioned this, yes I'll change this!! Let's get the Module requirements PR merged in and rebase this before making any more changes, then we can bump a major version!!